### PR TITLE
Fix jlmlocal and jlmthread tests

### PR DESCRIPTION
**Pull request by lumpfish**
Fixes https://github.com/eclipse/openj9/issues/1566

### CODE REVIEW

File: openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java
Comment (Mesbah-Alam) [inline openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java:1088]: Please remove the commented out line here.

File: openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java
Comment (Mesbah-Alam) [inline openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java:1136]: Should we fail the test in this case as well?

Comment (lumpfish) [reply openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java:1136]: Are you saying you don't think we should fail?  I can't see how we will ever know the condition is being hit unless the test fails.  The test pipelines would 'pass' so the detailed test logs (where the message is written) would never be examined.

The test thinks it has generated suitable arguments for the mxbean operation being invoked - so if that is not the case either something has changed (the argument values are no longer suitable) or there is a new mxbean operation / signature which the test is not (yet) aware of, so failing should be a good way to alert us to any such changes.

Comment (Mesbah-Alam) [reply openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java:1136]: I was wondering if we should add an Assert.fail(..) over there. Currently it's only printing the stack trace, but not failing the test, in that particular case.

Comment (lumpfish) [reply openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java:1136]: Ah - sorry, I assumed your comment applied to the code I had changed (where I added an Assert.fail() (line 1104).

I did not change th e code you mention - but since the RuntimeMBeanException is now caught in a try - catch block specific to the `invoke` method (line 1094) I believe the code in the `catch RuntimeMBeanException` block on line 1131 is now unreachable and should have been removed.  Please check if you agree - if so. I'll update the PR.

File: openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java
Comment (lumpfish) [reply openjdk.test.jlm/src/test.jlm/net/adoptopenjdk/test/jlm/resources/ThreadData.java:1089]: > Is the code inside this try block (line 1090) the only code that may generate a RuntimeMBeanException? If yes we can get rid of the catch block for RuntimeMBeanException further down (line 1131).

Yes.
I have removed the catch block at line 1131 and confirmed the tests still run OK: https://ci.adoptopenjdk.net/view/Test_grinder/job/Grinder/4185/

>  In that case, we should add an Assert.fail() inside the `if ( ! parameterTypesMatch )` block after line 1099.

I have deliberately called all overloaded methods even though the arguments we are supplying may be incorrect.  If the arguments are correct then the operation should succeed.  If the arguments are incorrect then `IllegalArgumentException` should be thrown.  The `parameterTypesMatch` check is there to enable the correct behavior to be determined.  So there is no need to fail the test if the `IllegalArgumentException` is thrown and that is what should happen.

**Comment by Mesbah-Alam** [comment]
Approving assuming the commented out line (https://github.com/AdoptOpenJDK/openjdk-systemtest/pull/364#discussion_r501892980) will be removed before merging.

**Review by Mesbah-Alam — APPROVED** [review]
LGTM

**Comment by lumpfish** [comment]
The commented out line highlighted in (#364 (comment)) now removed.

**Pull request merged**
